src/js/tabs/firmware_flasher.js

```diff
@@ -141,209 +137,35 @@ firmware_flasher.initialize = function (callback) {
             versions_e.empty();
             versions_e.append($(`<option value='0'>${i18n.getMessage("firmwareFlasherOptionLabelSelectFirmwareVersion")}</option>`));
 
-
-            const selectTargets = [];
-            Object.keys(builds)
+            Object.keys(targets)
                 .sort()
```

Comment:
```suggestion
                .sort((a,b) => a.target - b.target)
```

Sort works on the first part of the list now, but at the end there are still some legacy targets unsorted.